src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int capacity;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        Node<K, V> newNode = new Node<>(key, value);
+        putVal(newNode);
+    }
 
+    private void putVal(Node<K, V> newNode) {
+        if (table[Math.abs(newNode.hashCode % capacity)] == null) {
```

Comment:
you use this logic quite a few times, Why not extract it to a method like `getIndexFromHash`?